decoder/stats.go

```diff
@@ -63,8 +65,8 @@ type areaInvasionCountDetail struct {
 
 type areaQuestCountDetail struct {
 	count        int
-	pokemonCount [maxPokemonNo + 1]int
-	itemCount    [maxItemNo + 1]int
+	pokemonCount [maxPokemonNo + 1]map[int]int // for each pokemonId[megaCount] keep a count
```

Comment:
```suggestion
	pokemonDetails [maxPokemonNo + 1]map[int]int // for each pokemonId[megaCount] keep a count
```

Reply:
So for mega energy we need to add additional count for pokemonId? isn't that a item?

Reply:
mega energy is per pokemon type though (eg mega energy for bulbasaur).
It's like candy or xl candy for a pokemon type - which are also typed by pokemon and have an amount for the reward